lib/dry/system/plugins/zeitwerk.rb

```diff
@@ -0,0 +1,103 @@
+# frozen_string_literal: true
+
+require "dry/system/plugins/zeitwerk/compat_inflector"
+
+module Dry
+  module System
+    module Plugins
+      class Zeitwerk < Module
+        attr_reader :options
+
+        # @api private
+        def initialize(options)
+          @options = options
+          super()
+        end
```

Comment:
Would prefer that! Wasnt sure on the dry.rb policy here but I'll happily switch 👍🏻

Reply:
actually I think I remember why I did it now. by using the options, we can lazily `require 'zeitwerk'` for those that don't use it. Another way around this would be to register the plugin and require zeitwerk in a registration block. Happy to change if you think that's better.

Reply:
I took a stab at this and I realise there's enough _deferred_ default values based on the current state of the container that it doesn't make sense to use fixed kwargs and static defaults. Have left it as is :)